src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -95,7 +111,29 @@ internal sealed class VariableState
         /// </summary>
         private static readonly TypeSymbolWithAnnotations _invalidType = TypeSymbolWithAnnotations.Create(ErrorTypeSymbol.UnknownResultType);
 
-        private TypeSymbolWithAnnotations _resultType;
+        /// <summary>
+        /// The result and l-value type of the last visited expression.
+        /// Do no get or set directly
+        /// </summary>
+        private VisitResult _visitResult;
+
+        /// <summary>
+        /// The result type represents the state of the last visited expression.
+        /// </summary>
+        private TypeSymbolWithAnnotations ResultType
+        {
+            get => _visitResult.ResultType;
+            set
+            {
+                SetResult(resultType: value, lvalueType: value);
+            }
+        }
+
+        private void SetResult(TypeSymbolWithAnnotations resultType, TypeSymbolWithAnnotations lvalueType)
```

Comment:
Consider: SetResultType to line up with the property of the same name.

Reply:
`SetResultType` doesn't seem right. This method should be named the same as `SetResult(BoundExpression)`.
I think those two methods should be called `SetResult` or `SetVisitResult`.

---
In reply to: [258141377](https://github.com/dotnet/roslyn/pull/33447#discussion_r258141377) [](ancestors = 258141377)